src/FileFormats/LP/LP.jl

```diff
@@ -285,9 +285,545 @@ end
+    "binary" => Val{:binary},
+    "binaries" => Val{:binary},
+    "end" => Val{:quit},
+)
+
+const _SENSE_ALIAS = Dict(
+    "max" => :Max,
+    "maximize" => :Max,
+    "maximise" => :Max,
+    "maximum" => :Max,
+    "min" => :Min,
+    "minimize" => :Min,
+    "minimise" => :Min,
+    "minimum" => :Min,
+)
+
+const _SUBJECT_TO_ALIAS = ["subject to", "such that", "st", "s.t."]
+
+const _CONSTRAINT_SENSE = Dict(
+    "<" => :le,
+    "<=" => :le,
+    "=" => :eq,
+    "==" => :eq,
+    ">" => :ge,
+    ">=" => :ge,
+)
+
+function _verify_name(variable::String, maximum_length::Int)
+    if length(variable) > maximum_length
+        return false
+    end
+    # m = match(START_REG, variable)
+    # if !isnothing(m)
+    #     return false
+    # end
+    m = match(NAME_REG, variable)
+    if !isnothing(m)
+        return false
+    end
+    return true
+end
+
+struct _SOSConstraint
+    type::Int
+    weights::Vector{Float64}
+    columns::Vector{String}
+end
+
+# TODO Obj constant
+mutable struct TempLPModel
+    model_name::String
+    A::Vector{Vector{Tuple{Int,Float64}}}
+    c::Vector{Float64}
+    col_lower::Vector{Float64}
+    col_upper::Vector{Float64}
+    row_lower::Vector{Float64}
+    row_upper::Vector{Float64}
+    sense::Symbol
+    colcat::Vector{Symbol}
+    sos_constraints::Vector{_SOSConstraint}
+    col_to_name::Vector{String}
+    row_to_name::Vector{String}
+    open_constraint::Bool
+    maximum_length::Int
+    objective_constant::Float64
+    function TempLPModel(options::Options)
+        return new(
+            "",
+            Vector{Vector{Tuple{Int,Float64}}}[],
+            Float64[],
+            Float64[],
+            Float64[],
+            Float64[],
+            Float64[],
+            :Min,
+            Symbol[],
+            _SOSConstraint[],
+            String[],
+            String[],
+            false,
+            options.maximum_length,
+            0.0,
+        )
+    end
+end
+
+_set_sense!(T, data::TempLPModel, line) = nothing
+function _set_sense!(::Type{Val{:obj}}, data::TempLPModel, line)
+    return data.sense = _SENSE_ALIAS[lowercase(line)]
+end
+
+function _add_new_variable!(data::TempLPModel, name::String)
+    push!(data.col_lower, -Inf)
+    push!(data.col_upper, Inf)
+    push!(data.c, 0)
+    push!(data.colcat, :Cont)
+    push!(data.col_to_name, name)
+    return
+end
+
+function _get_variable_index!(data::TempLPModel, variable::String)
+    i = findfirst(isequal(variable), data.col_to_name)
+    if isnothing(i)
+        if !_verify_name(variable, data.maximum_length)
+            error("Invalid variable name $variable")
+        end
+        _add_new_variable!(data, variable)
+        return length(data.col_to_name)
+    end
+    return i
+end
+
+function _tokenize(line::AbstractString)
+    items = String.(split(line, " "))
+    return items[items.!=""]
+end
+
+function _parse_float(val::AbstractString)
+    if lowercase(val) == "-inf" || lowercase(val) == "-infinity"
```

Comment:
Not necessary just yet, but I saw in the MPS reader that calling `lowercase` was actually a significant bottleneck. We might want to call `lowercase(val)` once, rather than calling it 4 times (I don't think the compiler is smart enough to figure out it's the same computation with no side-effects).